### PR TITLE
Manifold DB Sync

### PR DESCRIPTION
**Pull request by nomadicoder**

### CODE REVIEW

File: manifold_airflow_dags/manifold_db_sync_dag.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+# We need to drop all connections to the manifold DB before we can drop the DB. But we don't want to drop the connection from the 
+# psql command doing the dropping or else the command fails.
+# So we
+# 1. drop connections to the db
+# 2. drop the db
+# 3. run pg_restore in a way that recreates the db and tables.
+# TODO - pg_restore returns some ignorable warnings that cause the command rc to be 2, which then fails in airflow
+disconnect_sql = """
+  sudo su - postgres bash -c 'echo "select pg_terminate_backend(pid) from pg_stat_activity where datname=\'manifold\' AND application_name!=\'psql\';" > disconnect_db.sql'
```
Comment (bibliotechy) [inline manifold_airflow_dags/manifold_db_sync_dag.py:91]: Two things
1. Multiline for readability please
1. Be more specific about where the `disconnect_db.sql` gets written. As of right now, I could not predict where to find it. Maybe put it in /tmp/disconnect_db.sql

**Review by bibliotechy — CHANGES_REQUESTED** [review]
Changes requested inline

File: manifold_airflow_dags/manifold_db_sync_dag.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    },
+    dag=DAG)
+
+POST_SLACK = PythonOperator(
+    task_id="slack_post_succ",
+    python_callable=slackpostonsuccess,
+    provide_context=True,
+    dag=DAG)
+
+# We use the restore user for this because the restore user does not exist in the prod DB
```
[the remark this answers is not in the document]
Comment (nomadicoder) [reply manifold_airflow_dags/manifold_db_sync_dag.py:79]: Move above `drop_and_restore_db`

File: manifold_airflow_dags/manifold_db_sync_dag.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+disconnect_sql = """
+  sudo su - postgres bash -c 'echo "select pg_terminate_backend(pid) from pg_stat_activity where datname=\'manifold\' AND application_name!=\'psql\';" > disconnect_db.sql'
+"""
+
+drop_and_restore_db = f"""
+sudo su - postgres bash -c \
+  "psql manifold -f disconnect_db.sql &&\
+  dropdb manifold &&\
+  createdb manifold &&\
+  pg_restore --dbname=manifold /tmp/%s &&\
```
[the remark this answers is not in the document]
Comment (nomadicoder) [reply manifold_airflow_dags/manifold_db_sync_dag.py:99]: Added `--user=restore`

File: manifold_airflow_dags/manifold_db_sync_dag.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    task_id="slack_post_succ",
+    python_callable=slackpostonsuccess,
+    provide_context=True,
+    dag=DAG)
+
+# We use the restore user for this because the restore user does not exist in the prod DB
+# This prevents us from accidentally running the restore on prod db.
+
+
+# We need to drop all connections to the manifold DB before we can drop the DB. But we don't want to drop the connection from the
```
[the remark this answers is not in the document]
Comment (nomadicoder) [reply manifold_airflow_dags/manifold_db_sync_dag.py:83]: Updated comments

File: manifold_airflow_dags/manifold_db_sync_dag.py (hunk 1 above)
Comment (nomadicoder) [reply manifold_airflow_dags/manifold_db_sync_dag.py:91]: Cleaned up comments.

Moved SQL script to /tmp/ with a clearer name.

File: manifold_airflow_dags/manifold_db_sync_dag.py [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+
+## 
+## BEFORE RUNNING, MAKE SURE YOU HAVE SET UP THE MANIFOLD_{server}_DB and MANIFOLD_{server}_APP connections
+## you'll need to run this locally. When targetting manifold vagrant, they should use conan as user, 
+## host of host.docker.internal, and the ports as defined in the manifold vagrant host file 
+## https://github.com/tulibraries/ansible-playbook-manifold/blob/qa/inventory/vagrant/hosts
+##  
+##
+for server in ["VAGRANT"]:
```
[the remark this answers is not in the document]
Comment (nomadicoder) [reply manifold_airflow_dags/manifold_db_sync_dag.py:127]: Replaced.

**Review by bibliotechy — APPROVED** [review]

**Review by dkinzer — APPROVED** [review]

**Pull request merged**